WebApi/DataModels/BarData.cs

```diff
@@ -34,10 +37,45 @@ public class BarData : IEquatable<BarData>
         [NotMapped]
         public double DistanceToCurrentLocation { get; set; }
 
-        // TODO : Make IComparable
-        public bool Equals(BarData other)
+        // Equals and GetHasCode are auto generated
+        public bool Equals(BarData y)
+        {
+            if (ReferenceEquals(this, y)) return true;
+            if (ReferenceEquals(this, null)) return false;
+            if (ReferenceEquals(y, null)) return false;
+            if (this.GetType() != y.GetType()) return false;
+            return string.Equals(this.Title, y.Title) && 
+                   string.Equals(this.BarId, y.BarId) && 
+                   this.AvgRating.Equals(y.AvgRating) && 
+                   this.UserRating == y.UserRating && 
+                   this.Latitude.Equals(y.Latitude) && 
+                   this.Longitude.Equals(y.Longitude) && 
+                   string.Equals(this.StreetAddress, y.StreetAddress) && 
+                   string.Equals(this.City, y.City) && 
+                   this.Categories == y.Categories && 
+                   Equals(this.UserRatings, y.UserRatings) && 
+                   Equals(this.Synonyms, y.Synonyms) && 
+                   this.DistanceToCurrentLocation.Equals(y.DistanceToCurrentLocation);
+        }
+
+        public int GetHashCode(BarData obj)
```

Comment:
Could you explain the purpose of this?

Reply:
Autogenerated code to compare two BarData objects. If all the statements are true, then the BarData objects are equal. 

Earlier it would only compare BarIDs which for now are names, so two different bar objects with same name would be chosen as equal.

Reply:
I meant the GetHashCode() method

Reply:
Sorry.
It's also autogenerated and I didn't delete it. [It seems that people on stack exchange recommend using it](https://stackoverflow.com/questions/371328/why-is-it-important-to-override-gethashcode-when-equals-method-is-overridden), but it's not needed in our case, so I can remove it.

Reply:
so I see. I think it will be okay if we leave it here.